apps/nsqd/nsqd.go

```diff
@@ -232,6 +238,10 @@ func (p *program) Start() error {
 	}
 	nsqd.Main()
 
+	// hook into addons
+	addons := contrib.NewEnabledNSQDAddons(opts.ModOpt, nsqd, nsqd.Logf)
```

Comment:
I moved the addon initialization to `Main()` after your initial comment and passed `lg.AppLogFunc` to the Addons, but then there was a circular dependency :p .  The only thing I could think of was to add an interface to represent NSQD?  Can you think of any easier ways?